Pod/Classes/NSAttributedString+BonMotUtilities.m

```diff
@@ -0,0 +1,61 @@
+//
+//  NSAttributedString+BonMotUtilities.m
+//  BonMot
+//
+//  Created by Eliot Williams on 3/9/16.
+//
+//
+
+#import "NSAttributedString+BonMotUtilities.h"
+#import "BONChain.h"
+#import "BONSpecial.h"
+
+@implementation NSAttributedString (BonMotUtilities)
+
+- (NSString *)humanReadableString
+{
+    NSString *originalString = self.string;
+    NSMutableString *composedHumanReadableString = [NSMutableString string];
+
+    [originalString enumerateSubstringsInRange:NSMakeRange(0, originalString.length) options:NSStringEnumerationByComposedCharacterSequences usingBlock:^(NSString *BONCNullable substring, NSRange substringRange, NSRange enclosingRange, BOOL *BONCNonnull stop) {
+
+        static NSCharacterSet *s_newLineCharacterSet = nil;
+        if (!s_newLineCharacterSet) {
+            s_newLineCharacterSet = [NSCharacterSet newlineCharacterSet];
+        }
+        NSCharacterSet *s_whiteSpaceAndNewLinesSet = nil;
+        if (!s_whiteSpaceAndNewLinesSet) {
+            s_whiteSpaceAndNewLinesSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
+        }
+
+        BONGeneric(NSDictionary, NSNumber *, NSString *)*specialCharacterSubstitutionNameDictionary = [BONSpecial humanReadableStringDictionary];
+        unichar character = [substring characterAtIndex:0];
+        NSString *specialCharacterSubstitutionString = specialCharacterSubstitutionNameDictionary[@(character)];
+
+        // Substitute BONSpecial Characters with @"{#name}"
+        if (specialCharacterSubstitutionString) {
+            [composedHumanReadableString appendFormat:@"%@", specialCharacterSubstitutionString];
+        }
+
+        // Substitute Images with @"{image#heightx#width}"
+        else if ([substring isEqualToString:BONSpecial.objectReplacementCharacter]) {
+            BONStringDict *attributes = [self attributesAtIndex:substringRange.location effectiveRange:NULL];
+            NSTextAttachment *attachment = attributes[NSAttachmentAttributeName];
+            UIImage *attachedImage = attachment.image;
+            NSString *imageSubstitutionString = [NSString stringWithFormat:@"{image%.0fx%.0f}", attachedImage.size.height, attachedImage.size.width];
```

Comment:
#? is it possible for an image to have a fractional size? If so, what are the possible values? And also if so, can/should we change this to reflect an integer number of pixels? I forget whether the image's `size` is in pixels or points.